pkg/apis/hive/v1alpha1/validating-webhooks/syncset_validating_admission_hook.go

```diff
@@ -233,14 +236,48 @@ func (a *SyncSetValidatingAdmissionHook) validateUpdate(admissionSpec *admission
 	}
 }
 
-func checkValidPatchTypes(patches []hivev1.SyncObjectPatch) (string, bool) {
-	validTypes := map[string]struct{}{"json": {}, "merge": {}, "strategic": {}}
-	for _, patch := range patches {
-		if _, ok := validTypes[patch.PatchType]; !ok {
-			return patch.PatchType, false
+func validatePatches(patches []hivev1.SyncObjectPatch, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	validPatchTypeSlice := []string{}
+	for k := range validPatchTypes {
+		validPatchTypeSlice = append(validPatchTypeSlice, k)
+	}
+
+	for i, patch := range patches {
+		if _, ok := validPatchTypes[patch.PatchType]; !ok {
+			allErrs = append(allErrs, field.NotSupported(fldPath.Index(i).Child("PatchType"), patch.PatchType,
+				validPatchTypeSlice))
 		}
 	}
-	return "", true
+	return allErrs
+}
+
+func validateResources(resources []runtime.RawExtension, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i, resource := range resources {
+		allErrs = append(allErrs, validateResource(resource, fldPath.Index(i))...)
+	}
+	return allErrs
+}
+
+func validateResource(resource runtime.RawExtension, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	rType := &metav1.TypeMeta{}
+	err := json.Unmarshal(resource.Raw, rType)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, resource.Raw, "Unable to unmarshal resource Kind and APIVersion"))
+		return allErrs
+	}
+
+	if invalid := invalidResourceGroupKinds[rType.GroupVersionKind().Group][rType.Kind]; invalid {
+		allErrs = append(allErrs,
+			field.Invalid(fldPath.Child("APIVersion"), rType.APIVersion,
+				fmt.Sprintf("Invalid resource APIVersion found: %s, use equivalent kubernetes group instead", rType.APIVersion)))
```

Comment:
Do not include the API version in the message string. It is already included in the output.
```
field.Invalid(fldPath.Child("APIVersion"), rType.APIVersion, "must use kubernetes group for this resource kind")
```